src/authentication/oauth2Auth.ts

```diff
@@ -167,6 +167,11 @@ export class Oauth2Auth extends AlfrescoApiClient {
             if (this.config.oauth2.silentLogin && !this.isPublicUrl()) {
                 this.implicitLogin();
             }
+
+            if(this.isValidAccessToken()) {
```

Comment:
Originally this piece of code block was in the else (negated) block of this condition:
```
!this.isValidToken() || !this.isValidAccessToken()
```
Shouldn't we add this in the new place:

```
this.isValidToken() && this.isValidAccessToken()
```

Not sure what is the difference between `isValidToken` and `isValidAccessToken`.

Reply:
What I don't understand why this code was moved from the "implicitLogin" method in the first place?
If it was just a refactoring to get rid of this side effect from the "implicitLogin", I think we need to add this negated condition back, don't we?

Reply:
Yep we can add this check more one is for the identity the other is for the access.